columnflow/production/cms/pdf.py

```diff
@@ -30,24 +30,62 @@
     # only run on mc
     mc_only=True,
 )
-def pdf_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
+def pdf_weights(
+    self: Producer,
+    events: ak.Array,
+    outlier_threshold: float = 0.5,
+    outlier_action: str = "ignore",
+    outlier_log_mode: str = "warning",
+    **kwargs,
+) -> ak.Array:
     """
     Producer that determines the pdf up and down variations on an event-by-event basis.
     This producer assumes that the nominal entry is always the first LHEPdfWeight value and
     that the nominal weight is already included in the LHEWeight.
     Can only be called with MC datasets.
 
+    The *outlier_action* defines the procedure of how to handle events with a pdf
+    uncertainty above the *outlier_threshold*. Supported modes are:
+
+        - ``"ignore"``: events are kept unmodified
+        - ``"remove"``: pdf weight nominal/up/down are all set to 0
+        - ``"raise"``: an exception is raised
+
+     Additionally, the verbosity of the procedure can be set with *outlier_log_mode*,
+     which offers the following options:
+
+        - ``"none"``: no message is given
+        - ``"info"``: a `logger.info` message is given
+        - ``"debug"``: a `logger.debug` message is given
+        - ``"warning"``: a `logger.warning` message is given
+
     Resources:
 
        - https://arxiv.org/pdf/1510.03865.pdf
     """
+
+    if not outlier_action:
+        outlier_action = "ignore"
+    if not outlier_log_mode:
+        outlier_log_mode = "none"
```

Comment:
Is this necessary? The defaults are already set in the function, right?

Reply:
Not really, I thought it might make sense to just allow something like False, None and empty strings, but you're probably right and we should just stick with the given (and documented) options